web/modules/custom/joinup_news/src/Controller/NewsController.php

```diff
@@ -51,8 +52,13 @@ public function add(RdfInterface $rdf_entity) {
   public function createNewsAccess(RdfInterface $rdf_entity) {
     // Check that the passed in RDF entity is a collection or a solution,
     // and that the user has the permission to create news.
-    if (in_array($rdf_entity->bundle(), ['collection', 'solution']) && $this->currentUser()->hasPermission('create rdf entity news')) {
-      return AccessResult::allowed();
+    if (in_array($rdf_entity->bundle(), ['collection', 'solution'])) {
+      if ($this->currentUser()->hasPermission('create rdf entity news')) {
+        return AccessResult::allowed();
+      }
+      if (OgAccess::userAccess($rdf_entity, 'create rdf entity news')->isAllowed()) {
+        return AccessResult::allowed();
+      }
```

Comment:
All this will be outdated and will start failing very soon when the latest changes in OG are merged in :) This can be as early as this evening.

It is also not the correct approach, this is checking a global permission, while it should check the per-group permission. For the moment this is fine though, the per-group permissions are not merged in OG yet. So this is for the moment the best solution we can do.

We should keep track of this technical debt though since it is known that this will become problematic in the future. Can you please make a followup ticket that we still need to fix this and put a @todo here with a link to the ticket? Thanks!